### ISSUE TITLE
needs addressable subsection/tasks

### ISSUE THREAD

**Issue by danwos** [useblocks/sphinxcontrib-needs#60]
A need might be quite big and its realisation/implementation may take place in different places inside the code.
So it would be great, if different parts of a need could be marked and addressed.

Ideas:
 * Targets based automatically on paragraphs. 
   Second paragraph could be target as :need:`my_id:2`
 * Targets based on manually set role, e.g.: `:need_task:'Do this...'`.
 * Targets based on manually set directive, e.g. `.. need_sub:: A lot of rst based content`
 * Targets based on sphinx sections
 * Nested Needs (inside content): A need inside a need inside a need 

An open point is also how to address this subsections from outside (number, hash, manual id, ...)

**Comment by danwos** [comment]
After discussion with @ubmarco, the second point 'manually set role' would be a helpful solution. 
So here is a short spec:

Inside any need, a role called :inline_need: (short :in:) can be used to give single sentences an own, unique id. E.g.:
```
:inline_need:`Top speed must be reached in 2 seconds.`
````

As for needs, an id gets automatically calculated by the role content and has the wrapping need as prefix/namespace.
E.g.: my_need_1.sub_1
Final output would be:`Top speed must be reached in 2 seconds (sub_1)`

But there must be also a way to set an id manually. The id of the wrapping need is still used as prefix. To do this use:
```
:inline_need:`(test_1) xyz`
```

If a given id starts with a /, no prefix id is used. This id must be unique comparing to ALL other needs.
```
:inline_need:`(/special_1)      xyz`
```
Whitespaces between id definition and starting text will get removed.

**Complex example**:
```
.. requirement:: Car must be awesome
   :id: my_car_1
   :tags: car
   :status: open

   My new car must be the fastest on the world. Therefor it shall have:

   * :inline_need:`(1)A topspeed of 300 km/h
   * :in:`(2)` An acceleration of 30 m/s`
   * :in:`(/awesome_1)` A turbo button`
```
`Inline needs` can be referenced in all common sphinx-needs link and reference functions.
```
Maybe :need:`my_car_1.2` will not work.

.. spec:: Build awesome car part 1
   :links: my_car_1.1, my_car1.2
```

Inside this spec `inline_needs` are not filterable.
There may be a feature request in future to support this.

**Technical details**
The whole role-content node shall get the id as reftarget.
The printed id at the end of the content-output should be a working link to the reftarget.

**Comment by danwos** [comment]
The new function is called now ``:need_part:``.
See docs of 0.3.0 for details.

Released with 0.3.0
